unit_tests/src/test_delay_hints_parser.cpp

```diff
@@ -75,7 +70,14 @@ auto make_double_until_success_or_until_time_string(int _nnnn,
 {
     return fmt::format("{}{} DOUBLE UNTIL SUCCESS OR UNTIL {}", _nnnn, _U, std::chrono::system_clock::to_time_t(_tp));
 }
-#endif
+
+template <typename Timepoint>
+auto make_double_until_success_or_upto_time_string(int _nnnn,
+                                                   const std::string_view _U,
+                                                   const Timepoint _tp) -> std::string
+{
+    return fmt::format("{}{} DOUBLE UNTIL SUCCESS UPTO {}", _nnnn, _U, std::chrono::system_clock::to_time_t(_tp));
```

Comment:
No `OR UPTO`?

The function name includes `_or_` and the delay directive above this uses it as well.